packages/ui/react-ui/src/hooks/useCrossMintModal.ts

```diff
@@ -0,0 +1,145 @@
+import { useState } from 'react';
+
+function createPopupString() {
+    return `height=750,width=400,left=${window.innerWidth / 2 - 200},top=${
+        window.innerHeight / 2 - 375
+    },resizable=yes,scrollbars=yes,toolbar=yes,menubar=true,location=no,directories=no, status=yes`;
+}
+
+interface IProps {
+    development: boolean;
+    clientId: string;
+    showOverlay: boolean;
+    crossmintOpened?: () => any,
+    crossmintClosed?: () => any,
+}
+
+interface IReturn {
+    connecting: boolean;
+    connect: (
+        collectionTitle?: string,
+        collectionDescription?: string,
+        collectionPhoto?: string,
+        mintTo?: string,
+        emailTo?: string,
+        listingId?: string
+    ) => void;
+}
+
+type MintQueryParams = {
+    clientId: string;
+    closeOnSuccess: string;
+    collectionTitle?: string;
+    collectionDescription?: string;
+    collectionPhoto?: string;
+    mintTo?: string;
+    emailTo?: string;
+    listingId?: string;
+}
+
+const PROD_URL = "https://www.crossmint.io";
+const DEV_URL = "http://localhost:3001";
+
+const executeIfExists = (fn?: () => any | undefined):void => {
```

Comment:
This file doesn't seem to have been run through prettier

Reply:
We dont have prettier set up in this repo I believe

Reply:
Oops, we should add it then. Mind running it manually for now?